gh-121249: unconditionally support `complex` types in `struct` (#132864)
python/cpython: Doc/library/struct.rst

+--------+--------------------------+--------------------+----------------+------------+
| ``P``  | :c:expr:`void \*`        | integer            |                | \(5)       |
+--------+--------------------------+--------------------+----------------+------------+

> | ``F``  | :c:expr:`float complex`  | complex            | 8              | \(10)      |

Reviewer: IMHO, now that the support is unconditional, the "F" and "D" rows should be listed after the "e" / "f" / "d" rows, this way all the floating types are together.

   For the ``'E'`` and ``'C'`` format characters, the packed representation uses
   the IEEE 754 binary32 and binary64 format for components of the complex
   number, regardless of the floating-point format used by the platform.

>    Note that support for complex types is available unconditionally,

Reviewer 2: Don't forget to update the blurb entry if needed.

Author: It looks one was already disbanded/merged to whatsnew.

Reviewer 2: It can be found in Misc/News.d/3.14.0a1.rst. The issue is that the latest changelog will contain "if the compiler has C11 complex arithmetic". Maybe we can just remove the NEWS entry from this file and add a new one with blurb?

Author: I edited that file.

>    Note, that support for complex types is available unconditionally,

Reviewer: Grammar fix
```suggestion
   Note that support for complex types is available unconditionally,
```

   regardless on support IEC 60559 compatible complex arithmetic (Annex G of

>    the C11 standard) by the compiler.  Each complex type interpreted as an

Reviewer: Grammar fix
```suggestion
   the C11 standard) by the compiler.  Each complex type is interpreted as an
```

>    array type containing exactly two elements of the corresponding real type

Reviewer 2: I don't really like "corresponding real type" followed by "real and imaginary parts". Maybe something  like:

```rst
Each complex type is interpreted as an array of two elements corresponding to the real and imaginary parts respectively and of corresponding C type (float or double).
```

Author: It's more or less a quote from the standard: "Each complex type has the same representation and alignment requirements as an array type containing exactly two elements of the corresponding real type; the first element is equal to the real part, and the second element to the imaginary part, of the complex number."

>    Note that support for complex types is available unconditionally,
>    regardless on support IEC 60559 compatible complex arithmetic (Annex G of
>    the C11 standard) by the compiler.  Each complex type is interpreted as an
>    array type containing exactly two elements of the corresponding real type
>    (real and imaginary parts, respectively).
> 

Author: ```suggestion
   Note that complex types (``F`` and ``D``) are available unconditionally,
   despite complex types being an optional feature in C.
   As specified in the C11 standard, each complex type is represented by a
   two-element C array containing, respectively, the real and imaginary parts.

```
